jetson/nav/rover.cpp

```diff
@@ -242,6 +261,47 @@ bool Rover::updateRover( RoverStatus newRoverStatus )
             mRoverStatus.odometry() = newRoverStatus.odometry();
             mRoverStatus.leftTarget() = newRoverStatus.leftTarget();
             mRoverStatus.rightTarget() = newRoverStatus.rightTarget();
+
+            // Perform basic caching, include special case when
+            // we store for the first time, but store an empty one
+            // so we don't just have "junk" in the target Cache.
+            if ( mRoverStatus.leftTarget().distance != -1 ) 
+            {
+                mRoverStatus.leftCacheTarget() = mRoverStatus.leftTarget();
+                mRoverStatus.getMisses() = 0;
+            }
+            else 
+            { // if this is empty, then so is rightTarget. increment misses
+                mRoverStatus.getMisses()++;
+            }
+
+            // Note: The only time this updates is ONLY if the leftTarget updates (by default
+            // the first target seen will be stored in leftTarget, and even if only 1 is seen
+            // this won't update, so we should be fine)
+            if ( mRoverStatus.rightTarget().distance != -1 ) 
```

Comment:
Move this if check to inside the check on line 268, so we know that this will only update if the left one updates.